Lib/test/test_typing.py

```diff
@@ -2802,6 +2802,14 @@ def __init__(self, attr: T) -> None:
         self.assertEqual(deepcopy(ci).attr, 1)
         self.assertEqual(ci.__orig_class__, C[int])
 
+    def test_parameter_propagation(self):
```

Comment:
I think this would fit better in a test class for ParamSpec. Also, it's worth testing that `__parameters__` is still right in more complicated cases, e.g. if the P is inside a Concatenate.

Reply:
I don't think one exists right now. I'm happy to create one. (`ParamSpecTestCase`?) Should that belong in its own file, or a separate class inside `test_typing.py`?

Reply:
There is a `ParamSpecTests` class.